anoncreds/protocol/prover.py

```diff
@@ -217,31 +232,37 @@ def proverId(self):
                 m2Tilde = cmod.integer(int(
                     nonRevocInitProof.TauListParams.m2)) if nonRevocInitProof else None
                 primaryInitProof = await self._primaryProofBuilder.initProof(
-                    schemaKey, c1, revealedAttrs, predicates,
-                    m1Tilde, m2Tilde)
+                    schemaId, c1, revealedAttrs, predicates,
+                    m1Tilde, m2Tilde, claim)
                 CList += primaryInitProof.asCList()
                 TauList += primaryInitProof.asTauList()
 
             initProof = InitProof(nonRevocInitProof, primaryInitProof)
-            initProofs[schemaKey] = initProof
+            initProofs[schemaId] = initProof
 
         # 2. hash
-        cH = self._get_hash(CList, TauList, nonce)
+        cH = self._get_hash(self._prepare_collection(CList), self._prepare_collection(TauList), nonce)
 
         # 3. finalize proofs
-        proofs = []
-        schemaKeys = []
-        for schemaKey, initProof in initProofs.items():
-            schemaKeys.append(schemaKey)
+        proofs = {}
+        for schemaId, initProof in initProofs.items():
             nonRevocProof = None
             if initProof.nonRevocInitProof:
                 nonRevocProof = await self._nonRevocProofBuilder.finalizeProof(
-                    schemaKey, cH, initProof.nonRevocInitProof)
+                    schemaId, cH, initProof.nonRevocInitProof)
             primaryProof = await self._primaryProofBuilder.finalizeProof(
-                schemaKey, cH, initProof.primaryInitProof)
-            proofs.append(Proof(primaryProof, nonRevocProof))
+                schemaId, cH, initProof.primaryInitProof)
 
-        return FullProof(cH, schemaKeys, proofs, CList)
+            schema = await self.wallet.getSchema(ID(schemaId=schemaId))
+
+            proof = Proof(primaryProof, nonRevocProof)
+            proofInfo = ProofInfo(proof=proof, schema_seq_no=schemaId, issuer_did=schema.issuerId)
+
+            proofs[str(schemaId)] = proofInfo
+
+        aggregatedProof = AggregatedProof(cH, self._prepare_collection(CList))
+
+        return FullProof(proofs, aggregatedProof, requestedProof)
```

Comment:
Where do we have a mapping between Proofs (ProofInfo) and Requested Attribute?
I think each requested proof (RequestedProof) should be actually a requested attribute, and contain the information like in https://github.com/hyperledger/indy-sdk/blob/master/src/api/anoncreds.rs#L658